iib/workers/tasks/opm_operations.py

```diff
@@ -738,3 +751,123 @@ def opm_create_empty_fbc(
         binary_image=binary_image,
         dockerfile_name='index.Dockerfile',
     )
+
+
+def opm_registry_add_fbc_fragment(
+    temp_dir: str,
+    from_index: str,
+    binary_image: str,
+    fbc_fragment: str,
+    overwrite_from_index_token: Optional[str],
+) -> None:
+    """
+    Add FBC fragment to from_index image.
+
+    This only produces the index.Dockerfile file and does not build the container image.
+
+    :param str temp_dir: the base directory to generate the database and index.Dockerfile in.
+    :param str from_index: the pull specification of the container image containing the index that
+        the index image build will be based from.
+    :param str binary_image: the pull specification of the container image where the opm binary
+        gets copied from. This should point to a digest or stable tag.
+    :param str fbc_fragment: the pull specification of fbc fragment to be added in from_index.
+    :pararm str overwrite_from_index_token: token used to access the image
+    """
+    fragment_path, fragment_operator = extract_fbc_fragment(
+        temp_dir=temp_dir, fbc_fragment=fbc_fragment
+    )
+
+    is_operator_in_db, index_db_path = verify_operator_exists(
+        from_index=from_index,
+        base_dir=temp_dir,
+        operator_package=fragment_operator,
+        overwrite_from_index_token=overwrite_from_index_token,
+    )
+
+    from_index_configs_dir = get_catalog_dir(from_index=from_index, base_dir=temp_dir)
+
+    log.info("The content of from_index located at %s", from_index_configs_dir)
+
+    if is_operator_in_db:
+        log.info('Removing %s from %s index.db ', fragment_operator, from_index)
```

Comment:
I almost feel the code under this `if` block should be a separate helper function. wdyt?

Reply:
I also had the same thoughts while writing this, but later realized that in the if block there was just removal and copying of operator packages, so thought its too few steps to split. Any ideas are more than welcome :)